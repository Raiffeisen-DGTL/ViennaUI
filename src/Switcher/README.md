# Switcher

Компонент выбора одного из множества или нескольких значений, в зависимости от целей

## Импорт

```
import { Switcher } from 'vienna-ui';
```

## Свойства / Props

| Prop | Type | Default | Description |
| --- | --- | --- | --- |
| ref | Ref<HTMLInputElement> \| undefined |  | Сcылка на нативный элемент input, доступна после отрисовки |
| size |ResponsiveProp<'m' \| 'l', Breakpoints> \| undefined
| | Размеры |
| onBlur | SwitcherEvent<FocusEvent<HTMLInputElement, Element>>  \| undefined | | Обработчик события при потере фокуса компонентом |
| onFocus | SwitcherEvent<FocusEvent<HTMLInputElement, Element>> \| undefined |  | Обработчик события при получении фокуса компонентом |
| onChange | SwitcherEvent<ChangeEvent<HTMLInputElement>>  \| undefined |  | Обработчик события при клике на компонент |
| checked | boolean \| undefined |  | Статус выбранного значения |
| name | string \| undefined |  |
| disabled | boolean \| undefined |  |


# Switch

Компонент выбора одного из множества или нескольких значений, в зависимости от целей.
`Switch` - компонент для переключения состояний. Он позволяет пользователю выбрать один из двух взамоисключающих вариантов, при этом один из них предустановлен.



```
    {() => {
        const [state, setState] = React.useState(false);
        return (
            <Switcher checked={state} onChange={({ value }) => setState(value)}>
                Switcher #1
            </Switcher>
        );
    }}
```

## Внешний вид

```
    {() => {
        const [state, setState] = React.useState({ first: false, second: true });
        return (
            <Groups>
                <Switcher
                    checked={state.first}
                    onChange={({ value }) => setState({ ...state, ...{ first: value } })}>
                    Switcher #1
                </Switcher>
                <Switcher
                    checked={state.second}
                    onChange={({ value }) => setState({ ...state, ...{ second: value } })}>
                    Switcher #2
                </Switcher>
            </Groups>
        );
    }}
```

#### Свойство checked

Свойство `checked` служит, чтобы контролировать состояние компонента.

```
    {() => {
        const [state, setState] = React.useState({ first: false, second: true });
        return (
            <Groups>
                <Switcher
                    checked={state.first}
                    onChange={({ value }) => setState({ ...state, ...{ first: value } })}>
                    Switcher #1
                </Switcher>
                <Switcher
                    checked={state.second}
                    onChange={({ value }) => setState({ ...state, ...{ second: value } })}>
                    Switcher #2
                </Switcher>
            </Groups>
        );
    }}
```

#### Без лейбла

```
    {() => {
        const [state, setState] = React.useState({ first: false, second: true });
        return (
            <Groups>
                <Switcher
                    checked={state.first}
                    onChange={({ value }) => setState({ ...state, ...{ first: value } })}
                />
                <Switcher
                    checked={state.second}
                    onChange={({ value }) => setState({ ...state, ...{ second: value } })}
                />
            </Groups>
        );
    }}
```

## Размеры

Компонент имеет стандартные размеры `m` (по-умолчанию) и `l`.

```
    {() => {
        const [state, setState] = React.useState({ first: false, second: true });
        return (
            <Groups>
                <Switcher
                    checked={state.first}
                    onChange={({ value }) => setState({ ...state, ...{ first: value } })}>
                    Switcher Size M
                </Switcher>
                <Switcher
                    size='l'
                    checked={state.second}
                    onChange={({ value }) => setState({ ...state, ...{ second: value } })}>
                    Switcher Size L
                </Switcher>
            </Groups>
        );
    }}
```

## Состояние disabled

Компонент имеет состояние 'disabled'.

```
    {() => {
        const [state, setState] = React.useState({ first: false, second: true });
        return (
            <Groups>
                <Switcher
                    disabled
                    checked={state.first}
                    onChange={({ value }) => setState({ ...state, ...{ first: value } })}>
                    Switcher #1
                </Switcher>
                <Switcher
                    disabled
                    checked={state.second}
                    onChange={({ value }) => setState({ ...state, ...{ second: value } })}>
                    Switcher #2
                </Switcher>
            </Groups>
        );
    }}
```

#### Адаптив

Для компонента Switcher, адаптив применяется к свойству `size`, что позволяет адаптивно менять размер компонента в зависимости от текущей ширины экрана. Для этого задайте свойству `size` объект вида `{ <breakpoint name>: <string value> }`

Основные значения breakpoints:

```
defaultBreakpoints = {
    s: 768,
    m: 1024,
    l: 1920,
    xl: 2560,
};

systemBreakpoints: Breakpoints = {
    /* xs */
    xs: `(max-width: ${defaultBreakpoints.s - 1}px)`,

    /* s */
    s: `(min-width: ${defaultBreakpoints.s}px)`,
    belowS: `(max-width: ${defaultBreakpoints.s - 1}px)`,

    /* m */
    m: `(min-width: ${defaultBreakpoints.m}px)`,
    belowM: `(max-width: ${defaultBreakpoints.m - 1}px)`,

    /* l */
    l: `(min-width: ${defaultBreakpoints.l}px)`,
    belowL: `(max-width: ${defaultBreakpoints.l - 1}px)`,

    /* xl */
    xl: `(min-width: ${defaultBreakpoints.xl}px)`,
};
```

```
    {() => {
        const [state, setState] = React.useState(false);
        return (
            <Switcher size={{ base: 'm', m: 'l' }} checked={state} onChange={({ value }) => setState(value)}>
                Adaptive switcher
            </Switcher>
        );
    }}
```

## Обработчики событий

Компонент поддерживает несколько обработчиков событий:

- `onBlur` - обработчик события при потере фокуса компонентом;
- `onFocus` - обработчик события при получении фокуса компонентом;
- `onChange` - обработчик события при клике на компонент;

```
    {() => {
        const [value, setValue] = React.useState(false);
        const handleChange = ({ value }) => setValue(value);
        return (
            <Switcher checked={value} onChange={handleChange}>
                Switcher
            </Switcher>
        );
    }}
```

## Состояние ViewOnly

Это состояние используется, когда нужно показать значение свитча без возможности изменения.
Может использоваться для построения форм, которые находятся в режиме просмотра, где все поля заполнены, но не доступны для редактирования.

Свойства:

- viewOnly - состояние `ViewOnly` (тип boolean);
- viewOnlyText - текст значения (тип ReactNode);

```
    <Switcher viewOnly checked={false}>Switcher</Switcher>
```