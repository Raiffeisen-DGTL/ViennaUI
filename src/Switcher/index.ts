export { Switcher, type SwitcherProps, type SwitcherEvent } from './Switcher';
