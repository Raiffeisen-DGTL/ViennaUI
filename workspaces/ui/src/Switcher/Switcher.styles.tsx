import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const switcher = getPresets('switcher', {
    base: null,
});

const slider = getPresets('switcher.slider', {
    base: null,
    size: 'size',
    acrive: null,
    checked: null,
    disabledCheked: null,
    disabledUnCheked: null,
});

const control = getPresets('switcher.control', {
    base: null,
    size: 'size',
    push: 'size',
    disabledCheked: null,
    disabledUnCheked: null,
});

const label = getPresets('switcher.label', {
    base: null,
    size: 'size',
    disabled: null,
});

export const Caption = styled.span<{ size: 'm' | 'l'; disabled: any }>`
    position: relative;
    display: inline-block;
    ${label.base};
    ${label.size};
    ${({ disabled }) => disabled && label.disabled}
`;

export const Box = styled.label<any>`
    display: inline-flex;
    ${switcher.base}
`;

export const Input = styled.input`
    position: absolute;
    left: -9999px;
`;

export const Handle = styled.span<{ size: 'm' | 'l'; disabled: any; checked: any }>`
    position: relative;
    display: inline-block;
    ${slider.base}
    ${slider.size}

    &:active {
        ${slider.active}
    }

    &::before {
        display: block;
        content: '';

        ${control.base}
        ${control.size}
    }

    ${({ checked }) =>
        checked &&
        css`
            ${slider.checked}

            &::before {
                ${control.push}
            }
        `}

    ${({ disabled, checked }): any =>
        disabled &&
        css`
            ${checked ? slider.disabledCheked : slider.disabledUnCheked}

            &::before {
                ${checked ? control.disabledCheked : control.disabledUnCheked}
            }
        `}
`;
