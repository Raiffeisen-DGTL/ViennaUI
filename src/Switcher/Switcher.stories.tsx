import React from 'react';
import { Story, Meta } from 'storybook';
import { Switcher, SwitcherProps } from './Switcher';

export default {
    title: 'Development/Switcher',
    component: Switcher,
} as Meta;

export const Overview: Story<SwitcherProps> = (args) => {
    const [value, setValue] = React.useState(false);
    const handleChange = ({ value }) => setValue(value);
    return (
        <Switcher {...args} checked={value} onChange={handleChange}>
            Switcher
        </Switcher>
    );
};

export const WithAdaptive: Story<SwitcherProps> = (args) => {
    const [value, setValue] = React.useState(false);
    const handleChange = ({ value }) => setValue(value);
    return (
        <Switcher size={{ m: 'm', s: 'l' }} {...args} checked={value} onChange={handleChange}>
            Switcher
        </Switcher>
    );
};

WithAdaptive.storyName = 'Адаптив';

export const ViewOnly: Story<SwitcherProps> = (args) => {
    return (
        <Switcher viewOnly checked={false}>
            Switcher
        </Switcher>
    );
};
