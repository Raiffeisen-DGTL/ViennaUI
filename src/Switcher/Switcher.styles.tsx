import styled, { css } from 'styled-components';
import { getPresets } from '../Utils/styling';
import { responsivePreset } from '../Utils/responsiveness';
import { Size } from './Switcher';

const switcher = getPresets('switcher', {
    base: null,
});

const slider = getPresets('switcher.slider', {
    base: null,
    size: responsivePreset('$size', 'm'),
    active: null,
    checked: null,
    disabledCheked: null,
    disabledUnCheked: null,
});

const control = getPresets('switcher.control', {
    base: null,
    size: responsivePreset('$size', 'm'),
    push: responsivePreset('$size', 'm'),
    disabledCheked: null,
    disabledUnCheked: null,
});

const label = getPresets('switcher.label', {
    base: null,
    size: responsivePreset('$size', 'm'),
    disabled: null,
});

interface PropsCaption {
    $size?: Size;
    $disabled?: boolean;
}
export const Caption = styled.span<PropsCaption>`
    position: relative;
    display: inline-block;
    ${label.base};
    ${label.size};
    ${({ $disabled }) => $disabled && label.disabled}
`;

export const Box = styled.label`
    display: inline-flex;
    ${switcher.base}
`;

export const Input = styled.input`
    position: absolute;
    left: -9999px;
`;

interface PropsHandle {
    $size?: Size;
    $disabled?: boolean;
    $checked?: boolean;
}
export const Handle = styled.span<PropsHandle>`
    position: relative;
    display: inline-block;
    ${slider.base}
    ${slider.size}

    &:active {
        ${slider.active}
    }

    &::before {
        display: block;
        content: '';

        ${control.base}
        ${control.size}
    }

    ${({ $checked }) =>
        $checked &&
        css`
            ${slider.checked}

            &::before {
                ${control.push}
            }
        `}

    ${({ $disabled, $checked }) =>
        $disabled &&
        css`
            ${$checked ? slider.disabledCheked : slider.disabledUnCheked}

            &::before {
                ${$checked ? control.disabledCheked : control.disabledUnCheked}
            }
        `}
`;
