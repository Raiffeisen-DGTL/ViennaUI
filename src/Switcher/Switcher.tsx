import React, {
    useCallback,
    ForwardRefExoticComponent,
    RefAttributes,
    FocusEvent,
    ChangeEvent,
    Ref,
    FormEvent,
} from 'react';
import { Box, Input, Handle, Caption } from './Switcher.styles';
import { ResponsiveProp, Breakpoints } from '../Utils/responsiveness';
import { BoxStyled } from '../Utils/styled';

export type Size<B = Breakpoints> = ResponsiveProp<'m' | 'l', B>;

export type SwitcherEvent<T> = (event: T, data?: { name?: string; value: boolean }) => void;

export interface SwitcherProps<B = Breakpoints>
    extends Omit<BoxStyled<typeof Box, {}>, 'onFocus' | 'onBlur' | 'onChange'> {
    /** Размеры */
    size?: Size<B>;
    /** Имя компонента (важно для группировки) */
    name?: string;
    /** Состояние компонента */
    checked?: boolean;
    /** Компонент неактивен если true */
    disabled?: boolean;
    /** Обработчик события при потере фокуса компонентом  */
    onBlur?: SwitcherEvent<FocusEvent<HTMLInputElement>>;
    /** Обработчик события при получении фокуса компонентом  */
    onFocus?: SwitcherEvent<FocusEvent<HTMLInputElement>>;
    /** Обработчик события при клике на компонент  */
    onChange?: SwitcherEvent<ChangeEvent<HTMLInputElement>>;
}

export const Switcher = React.forwardRef(
    <B,>(props: SwitcherProps<B extends void ? Breakpoints : B>, ref: Ref<HTMLInputElement>) => {
        const {
            children,
            size = 'm',
            checked = false,
            disabled,
            name,
            tabIndex,
            onChange,
            onBlur,
            onFocus,
            ...attrs
        } = props;

        const handleEvent = useCallback(
            (callback) => {
                return (event: FormEvent<HTMLInputElement>) => {
                    if (!disabled && typeof callback === 'function') {
                        callback(event, { name, value: event.currentTarget.checked });
                    }
                };
            },
            [disabled, name]
        );

        return (
            <Box {...(attrs as {})}>
                <Handle tabIndex={0} $size={size} $disabled={disabled} $checked={checked} />
                <Input
                    ref={ref}
                    tabIndex={tabIndex}
                    name={name}
                    type='checkbox'
                    checked={checked}
                    disabled={disabled}
                    onChange={handleEvent(onChange)}
                    onBlur={handleEvent(onBlur)}
                    onFocus={handleEvent(onFocus)}
                />
                {children ? (
                    <Caption $size={size} $disabled={disabled}>
                        {children}
                    </Caption>
                ) : null}
            </Box>
        );
    }
) as ForwardRefExoticComponent<SwitcherProps & RefAttributes<HTMLInputElement>>;

Switcher.displayName = 'Switcher';
