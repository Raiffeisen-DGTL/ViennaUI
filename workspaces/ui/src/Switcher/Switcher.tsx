import React, { useCallback, HTMLAttributes } from 'react';
import { Box, Input, Handle, Caption } from './Switcher.styles';

export type SwitcherEvent<T> = (event: T, data?: { name?: string; value: boolean }) => void;
export interface SwitcherProps
    extends Omit<HTMLAttributes<HTMLDivElement>, 'size' | 'onChange' | 'onBlur' | 'onFocus'> {
    /** Отображаемый лейбл */
    children?: React.ReactNode;
    /** Сcылка на нативный элемент input, доступна после отрисовки */
    ref?: React.Ref<HTMLInputElement>;
    /** Размеры */
    size?: 'm' | 'l';
    /** Имя компонента (важно для группировки) */
    name?: string;
    /** Состояние компонента */
    checked?: boolean;
    /** Компонент неактивен если true */
    disabled?: boolean;
    /** Обработчик события при потере фокуса компонентом  */
    onBlur?: SwitcherEvent<React.FocusEvent<HTMLInputElement>>;
    /** Обработчик события при получении фокуса компонентом  */
    onFocus?: SwitcherEvent<React.FocusEvent<HTMLInputElement>>;
    /** Обработчик события при клике на компонент  */
    onChange?: SwitcherEvent<React.ChangeEvent<HTMLInputElement>>;
}

export const Switcher: React.FC<SwitcherProps> = React.forwardRef(
    (props: React.PropsWithChildren<SwitcherProps>, ref: React.Ref<HTMLInputElement>): JSX.Element => {
        const { children, size = 'm', checked, disabled, name, tabIndex, onChange, onBlur, onFocus, ...attrs } = props;

        const handleEvent = useCallback(
            (callback: SwitcherEvent<any> | undefined) => {
                return (event: React.FormEvent<HTMLInputElement>) => {
                    if (!disabled && typeof callback === 'function') {
                        callback(event, { name, value: event.currentTarget.checked });
                    }
                };
            },
            [disabled, name]
        );

        return (
            <Box size={size} disabled={disabled} {...attrs}>
                <Handle size={size} disabled={disabled} checked={checked} />
                <Input
                    ref={ref}
                    tabIndex={tabIndex}
                    name={name}
                    type='checkbox'
                    checked={checked}
                    disabled={disabled}
                    onChange={handleEvent(onChange)}
                    onBlur={handleEvent(onBlur)}
                    onFocus={handleEvent(onFocus)}
                />
                {children ? (
                    <Caption size={size} disabled={disabled}>
                        {children}
                    </Caption>
                ) : null}
            </Box>
        );
    }
);

Switcher.defaultProps = {
    size: 'm',
    checked: false,
};

Switcher.displayName = 'Switcher';
export default Switcher;
