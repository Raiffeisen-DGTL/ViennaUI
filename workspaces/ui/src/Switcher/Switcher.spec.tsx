import React from 'react';
import Switcher, { SwitcherProps } from './Switcher';

test('Switcher', () => {
    const snap = snapshot.render(<Switcher />);
    expect(snap).toMatchSnapshot();
});

test('Switcher w/ label', () => {
    const snap = snapshot.render(<Switcher>Label</Switcher>);
    expect(snap).toMatchSnapshot();
});

test('Switcher w/ name', () => {
    const snap = snapshot.render([
        <Switcher key='1' name='switcher1' />,
        <Switcher key='2' name='switcher2'>
            Label
        </Switcher>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Switcher w/ checked', () => {
    const snap = snapshot.render([
        <Switcher key='1' checked />,
        <Switcher key='2' checked>
            Label
        </Switcher>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Switcher w/ disabled', () => {
    const snap = snapshot.render([
        <Switcher key='1' disabled />,
        <Switcher key='2' disabled>
            Label
        </Switcher>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Switcher w/ checked and disabled', () => {
    const snap = snapshot.render([
        <Switcher key='1' checked disabled />,
        <Switcher key='2' checked disabled>
            Label
        </Switcher>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Switcher w/ size', () => {
    const sizes: SwitcherProps['size'][] = ['m', 'l'];

    sizes.forEach((size) => {
        const snap = snapshot.render(<Switcher size={size}>Label</Switcher>);
        expect(snap).toMatchSnapshot(size);
    });
});

test('Switcher w/ any props', () => {
    const props = {
        className: 'className',
        id: 'id',
        title: 'title',
        onClick: () => 'Click',
    };

    const component = (
        <Switcher {...props} name='name'>
            Label
        </Switcher>
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.shallow(component);
    for (const [key, value] of Object.entries(props)) {
        expect(wrapper.props()[key]).toEqual(value);
    }
});

test('Switcher w/ onBlur()', () => {
    const fn = jest.fn();
    const event = { currentTarget: { checked: true } };

    const snap = snapshot.shallow(
        <Switcher name='switcher' checked onBlur={fn}>
            Label
        </Switcher>
    );

    snap.childAt(1).simulate('blur', event);
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { name: 'switcher', value: true });
});

test('Switcher w/ onFocus()', () => {
    const fn = jest.fn();
    const event = { currentTarget: { checked: false } };

    const snap = snapshot.shallow(
        <Switcher name='switcher' onFocus={fn}>
            Label
        </Switcher>
    );

    snap.childAt(1).simulate('focus', event);
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { name: 'switcher', value: false });
});

test('Switcher w/ onChange()', () => {
    const fn = jest.fn();
    const event = { currentTarget: { checked: true } };

    const snap = snapshot.shallow(
        <Switcher name='switcher' onChange={fn}>
            Label
        </Switcher>
    );

    snap.childAt(1).simulate('change', event);
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { name: 'switcher', value: true });
});

test('Switcher w/ onChange() disabled', () => {
    const fn = jest.fn();
    const event = { currentTarget: { checked: true } };

    const snap = snapshot.shallow(
        <Switcher name='switcher' disabled onChange={fn}>
            Label
        </Switcher>
    );

    snap.childAt(1).simulate('change', event);
    expect(fn).not.toHaveBeenCalled();
    expect(fn).not.toHaveBeenCalledWith(event, { name: 'switcher', value: true });
});
