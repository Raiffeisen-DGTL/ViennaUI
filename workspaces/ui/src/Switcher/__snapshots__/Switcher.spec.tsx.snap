// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher 1`] = `
<label
  class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
>
  <span
    class="Switcherstyles__Handle-nl4ecl-3 eiRxnP"
  />
  <input
    class="Switcherstyles__Input-nl4ecl-2 ePUicI"
    type="checkbox"
  />
</label>
`;

exports[`Switcher w/ any props 1`] = `
<label
  class="Switcherstyles__Box-nl4ecl-1 hnUnvl className"
  id="id"
  title="title"
>
  <span
    class="Switcherstyles__Handle-nl4ecl-3 eiRxnP"
  />
  <input
    class="Switcherstyles__Input-nl4ecl-2 ePUicI"
    name="name"
    type="checkbox"
  />
  <span
    class="Switcherstyles__Caption-nl4ecl-0 cydKQj"
  >
    Label
  </span>
</label>
`;

exports[`Switcher w/ checked 1`] = `
Array [
  <label
    class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
  >
    <span
      checked=""
      class="Switcherstyles__Handle-nl4ecl-3 fBkrAL"
    />
    <input
      checked=""
      class="Switcherstyles__Input-nl4ecl-2 ePUicI"
      type="checkbox"
    />
  </label>,
  <label
    class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
  >
    <span
      checked=""
      class="Switcherstyles__Handle-nl4ecl-3 fBkrAL"
    />
    <input
      checked=""
      class="Switcherstyles__Input-nl4ecl-2 ePUicI"
      type="checkbox"
    />
    <span
      class="Switcherstyles__Caption-nl4ecl-0 cydKQj"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Switcher w/ checked and disabled 1`] = `
Array [
  <label
    class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
    disabled=""
  >
    <span
      checked=""
      class="Switcherstyles__Handle-nl4ecl-3 buMWEi"
      disabled=""
    />
    <input
      checked=""
      class="Switcherstyles__Input-nl4ecl-2 ePUicI"
      disabled=""
      type="checkbox"
    />
  </label>,
  <label
    class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
    disabled=""
  >
    <span
      checked=""
      class="Switcherstyles__Handle-nl4ecl-3 buMWEi"
      disabled=""
    />
    <input
      checked=""
      class="Switcherstyles__Input-nl4ecl-2 ePUicI"
      disabled=""
      type="checkbox"
    />
    <span
      class="Switcherstyles__Caption-nl4ecl-0 ezLBpR"
      disabled=""
    >
      Label
    </span>
  </label>,
]
`;

exports[`Switcher w/ disabled 1`] = `
Array [
  <label
    class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
    disabled=""
  >
    <span
      class="Switcherstyles__Handle-nl4ecl-3 dHTjdo"
      disabled=""
    />
    <input
      class="Switcherstyles__Input-nl4ecl-2 ePUicI"
      disabled=""
      type="checkbox"
    />
  </label>,
  <label
    class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
    disabled=""
  >
    <span
      class="Switcherstyles__Handle-nl4ecl-3 dHTjdo"
      disabled=""
    />
    <input
      class="Switcherstyles__Input-nl4ecl-2 ePUicI"
      disabled=""
      type="checkbox"
    />
    <span
      class="Switcherstyles__Caption-nl4ecl-0 ezLBpR"
      disabled=""
    >
      Label
    </span>
  </label>,
]
`;

exports[`Switcher w/ label 1`] = `
<label
  class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
>
  <span
    class="Switcherstyles__Handle-nl4ecl-3 eiRxnP"
  />
  <input
    class="Switcherstyles__Input-nl4ecl-2 ePUicI"
    type="checkbox"
  />
  <span
    class="Switcherstyles__Caption-nl4ecl-0 cydKQj"
  >
    Label
  </span>
</label>
`;

exports[`Switcher w/ name 1`] = `
Array [
  <label
    class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
  >
    <span
      class="Switcherstyles__Handle-nl4ecl-3 eiRxnP"
    />
    <input
      class="Switcherstyles__Input-nl4ecl-2 ePUicI"
      name="switcher1"
      type="checkbox"
    />
  </label>,
  <label
    class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
  >
    <span
      class="Switcherstyles__Handle-nl4ecl-3 eiRxnP"
    />
    <input
      class="Switcherstyles__Input-nl4ecl-2 ePUicI"
      name="switcher2"
      type="checkbox"
    />
    <span
      class="Switcherstyles__Caption-nl4ecl-0 cydKQj"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Switcher w/ size: l 1`] = `
<label
  class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
>
  <span
    class="Switcherstyles__Handle-nl4ecl-3 cDhOpV"
  />
  <input
    class="Switcherstyles__Input-nl4ecl-2 ePUicI"
    type="checkbox"
  />
  <span
    class="Switcherstyles__Caption-nl4ecl-0 bnhcym"
  >
    Label
  </span>
</label>
`;

exports[`Switcher w/ size: m 1`] = `
<label
  class="Switcherstyles__Box-nl4ecl-1 hnUnvl"
>
  <span
    class="Switcherstyles__Handle-nl4ecl-3 eiRxnP"
  />
  <input
    class="Switcherstyles__Input-nl4ecl-2 ePUicI"
    type="checkbox"
  />
  <span
    class="Switcherstyles__Caption-nl4ecl-0 cydKQj"
  >
    Label
  </span>
</label>
`;
